cfme/configure/settings.py

```diff
@@ -1,173 +1,228 @@
+
+
+@attr.s
+class TimeProfile(Updateable, BaseEntity):
+
+    description = attr.ib(default=None)
+    scope = attr.ib(default=None)
+    days = attr.ib(default=True)
+    hours = attr.ib(default=True)
+    timezone = attr.ib(default=None)
 
     def update(self, updates):
+        """
+        This method is used for updating the time_profile
+
+        Args:
+            updates: It the object of the time_profile that we need to update.
+        """
         view = navigate_to(self, 'Edit')
-        changed = view.timeprofile_form.fill({
+        changed = view.form.fill({
             'description': updates.get('description'),
             'scope': updates.get('scope'),
             'days': updates.get('days'),
             'hours': updates.get('hours'),
             'timezone': updates.get('timezone'),
         })
         if changed:
-            view.timeprofile_form.save_edit_button.click()
-            view.flash.assert_message(
-                'Time Profile "{}" was saved'.format(updates.get('description', self.description)))
+            view.form.save.click()
+            view.flash.assert_no_error()
 
-    def copy(self, name=None):
-        view = navigate_to(self, 'Copy')
+    def copy(self, description=None, cancel=False):
+        """
+        This method performs the copy of the provided time profile object.
+        Args:
+            description (str) : It's the descriptive name of the new copied time_profile.
+            cancel (bool) : This variable performs cancel operation while copy.
+
+        return: It returns the object of the copied time_profile.
+        """
 
-        if name is not None:
-            new_timeprofile = Timeprofile(description=name, scope=self.scope)
-            changed = view.timeprofile_form.fill({
-                'description': name,
+        view = navigate_to(self, 'Copy')
+        if description is not None:
+            new_time_profile = self.parent.instantiate(description=description, scope=self.scope)
+            changed = view.form.fill({
+                'description': description,
                 'scope': self.scope,
             })
         else:
-            new_timeprofile = Timeprofile(description="{} copy".format(self.description),
-                                          scope=self.scope)
-            changed = view.timeprofile_form.fill({
+            new_time_profile = self.parent.instantiate(
+                description="{} copy".format(self.description),
+                scope=self.scope
+            )
+            changed = view.form.fill({
                 'description': "{} copy".format(self.description),
                 'scope': self.scope,
             })
 
-        if changed:
-            view.timeprofile_form.save_button.click()
-        return new_timeprofile
+        if not cancel and changed:
+            view.form.add.click()
+            view.flash.assert_no_error()
+        return new_time_profile
```

Comment:
**REQUIRED**
You return a TimeProfile copy even if "Cancel" button was clicked or the form was not changed. Does it make sense to return the copy in such cases?

Reply:
Nice catch